devtools/working-with-dsproxy/working-with-dsproxy.md

```diff
@@ -0,0 +1,488 @@
+}
+```
+
+#### Wipe Dai debt from the CDP
+
+Wipe debt of the CDP with DAI and pay the stability fee with MKR available on the DSProxy contract.
+
+```text
+tub.wipe(cup, wad);
+```
+
+Before we proceed to the next section of this guide, please ensure your code matches the `WipeProxy` contract below
+
+```text
+contract WipeProxy is DSMath {
+    function setAllowance(TokenLike token_, address spender_) private {
+        if (token_.allowance(address(this), spender_) != uint(-1)) {
+            token_.approve(spender_, uint(-1));
+        }
+    }
+
+    function wipeWithDai(
+        address _tub,
+        address _DAIExchange,
+        address _MKRExchange,
+        uint cupid,
+        uint wad
+    ) public {
+        require(wad > 0);
+
+        TubLike tub = TubLike(_tub);
+        UniswapExchangeLike daiEx = UniswapExchangeLike(_DAIExchange);
+        UniswapExchangeLike mkrEx = UniswapExchangeLike(_MKRExchange);
+        TokenLike dai = tub.sai();
+        TokenLike mkr = tub.gov();
+        PepLike pep =   tub.pep();
+
+        bytes32 cup = bytes32(cupid);
+
+        setAllowance(dai, _tub);
+        setAllowance(mkr, _tub);
+        setAllowance(dai, _DAIExchange);
+
+        (bytes32 val, bool ok) = pep.peek();
+
+        // MKR required for wipe = Stability fees accrued in Dai / MKRUSD value
+        uint mkrFee = wdiv(rmul(wad, rdiv(tub.rap(cup), tub.tab(cup))), uint(val));
+
+        uint ethAmt = mkrEx.getEthToTokenOutputPrice(mkrFee);
+        uint daiAmt = daiEx.getTokenToEthOutputPrice(ethAmt);
+
+        daiAmt = add(wad, daiAmt);
+        require(dai.transferFrom(msg.sender, address(this), daiAmt));
+
+        if(ok && val != 0) {
+           daiEx.tokenToTokenSwapOutput(mkrFee, daiAmt, uint(999000000000000000000), uint(1645118771), address(mkr));
+        }
+
+        tub.wipe(cup, wad);
+    }
+}
+
+```
+
+### Deployment and Execution
+
+Before we begin, ensure you have some Kovan ETH to pay gas for transactions and Kovan Dai on the address by following instructions on this [guide](https://github.com/makerdao/developerguides/blob/master/dai/dai-token/dai-token.md#testnet)
+
+Build the `wipe-proxy` project
+
+```bash
+dapp build
+```
+
+Deploy the WipeProxy contract
+
+```bash
+dapp create WipeProxy
+```
+
+Make a note of the contract address returned after successful execution and store it as a variable
+
+```bash
+export WIPEPROXY=0xfd92bd57d369714f519c3e6095d62d5872114e34
+```
+
+Deploy your own DSProxy contract for your address using the factory contract present on Kovan
+
+```bash
+seth send 0x64a436ae831c1672ae81f674cab8b6775df3475c 'build()'
+```
+
+This transaction might fail if you already have deployed a DSProxy contract before from this address. You can check if you have one now with this command
+
+```bash
+seth call 0x64a436ae831c1672ae81f674cab8b6775df3475c 'proxies(address)(address)' 0xYourAddressHere
+```
+
+Make a note of the returned DSProxy contract address to use in a command later.
```

Comment:
This would be a great point to set a variable that can be used in the call to `execute( )`
Ex: `Let's set the returned DSProxy contract address to be used later:
export MyProxy= 0xYourReturnedProxyAddress